taichi/ir/ir.cpp

```diff
@@ -482,6 +482,10 @@ void DelayedIRModifier::replace_with(Stmt *stmt,
   to_replace_with.emplace_back(stmt, std::move(new_statements), replace_usages);
 }
 
+void DelayedIRModifier::extract_to_block_front(Stmt *stmt, Block *blk) {
+  to_extract_to_block_front.emplace_back(stmt, blk);
+}
+
 bool DelayedIRModifier::modify_ir() {
   bool force_modified = modified_;
```

Comment:
nit: do we need to update `force_modified` here?

Reply:
no :-)